src/Build/BuildCheck/Infrastructure/BuildCheckForwardingLogger.cs

```diff
@@ -7,13 +7,61 @@
 using System.Text;
 using System.Threading.Tasks;
 using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Experimental.BuildCheck.Acquisition;
+using Microsoft.Build.Framework;
+using static Microsoft.Build.Experimental.BuildCheck.Infrastructure.BuildCheckManagerProvider;
 
 namespace Microsoft.Build.Experimental.BuildCheck.Infrastructure;
 
 /// <summary>
 /// Forwarding logger for the build check infrastructure.
-/// For now we jus want to forward all events, while disable verbose logging of tasks.
+/// For now we just want to forward all events that are needed for BuildCheckConnectorLogger and filter out all other.
+/// If the custom analyzer is detected, starts to unconditionally forward all events.
 /// In the future we may need more specific behavior.
 /// </summary>
-internal class BuildCheckForwardingLogger : CentralForwardingLogger
-{ }
+/// <remarks>
+/// Ensure that events filtering is in sync with <see cref="BuildCheckConnectorLogger"/>
+/// </remarks>
+internal class BuildCheckForwardingLogger : IForwardingLogger
+{
+    public IEventRedirector? BuildEventRedirector { get; set; }
+
+    public int NodeId { get; set; }
+
+    public LoggerVerbosity Verbosity { get => LoggerVerbosity.Quiet; set { return; } }
```

Comment:
Is the `set` used somewhere and should it be overall? If we expect only `LoggerVerbosity.Quiet`. 
Also could you please clarify why we are returning const value here?

Reply:
For this forwarding logger there is not much point in setting the verbosity, but when initializing the logging system will try to set some value there, so we need the set function to be defined. As for the `LoggerVerbosity.Quiet` - there are some checks that check the maximum verbosity of all the loggers that are attached and we would not want this logger to affect these checks, hence - lowest possible verbosity is assigned to this logger.